### PR TITLE
feat(java): Add java SDK for client sdks

### PR DESCRIPTION
**Pull request by yquansah**
Adding Java client SDK. Closes: https://github.com/flipt-io/flipt-client-sdks/issues/81

Completes FLI-771

### CODE REVIEW

File: flipt-client-java/.idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
```
Comment (markphelps) [inline flipt-client-java/.idea/.gitignore:2]: can we exclude all `.idea/` since its editor specific?

File: flipt-client-java/build.gradle [hunk 2]
```diff
@@ -0,0 +1,86 @@
+plugins {
+    id 'java-library'
+    id 'maven-publish'
+    id "com.diffplug.spotless" version "6.25.0"
+}
+
+group = 'io.flipt'
+version = '1.0.0'
```
Comment (markphelps) [inline flipt-client-java/build.gradle:8]:
```suggestion
version = '0.1.0'
```

File: flipt-client-java/src/main/java/io/flipt/client/models/EngineOpts.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    this.reference = reference;
+  }
+
+  @JsonProperty("url")
+  public Optional<String> getUrl() {
+    return url;
+  }
+
+  @JsonProperty("update_interval")
+  public Optional<Integer> getUpdateInterval() {
```
Comment (markphelps) [inline flipt-client-java/src/main/java/io/flipt/client/models/EngineOpts.java:31]: Can we make this a `Duration` ? https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html

File: flipt-client-java/src/main/java/io/flipt/client/models/EvalRequest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package io.flipt.client.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+
+public class EvalRequest {
```
Comment (markphelps) [inline flipt-client-java/src/main/java/io/flipt/client/models/EvalRequest.java:6]: can we make this private or protected? I don't think we want to expose this in the public API

File: flipt-client-java/build.gradle [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+    if (osName.contains("linux") || osName.contains("nix")) {
+        String archName = System.getProperty("os.arch").toLowerCase()
+        if (archName == "amd64" || archName == "x86_64") {
+            systemProperty("jna.library.path", System.getProperty("user.dir")+File.separator+"lib"+File.separator+"ext"+File.separator+"linux_x86_64")
+        }
+        else if (archName == "aarch64" || archName == "arm64") {
+            systemProperty("jna.library.path", System.getProperty("user.dir")+File.separator+"lib"+File.separator+"ext"+File.separator+"linux_arm64")
+        }
+    }
```
Comment (markphelps) [inline flipt-client-java/build.gradle:85]: is it possible to make this fail the build if none of these are true?

**Review by markphelps — COMMENTED** [review]
couple minor things. looks great!! thank you for tackling this. you are the java expert now on the team

File: flipt-client-java/src/main/java/io/flipt/client/models/EvalRequest.java (hunk 4 above)
Comment (yquansah) [reply flipt-client-java/src/main/java/io/flipt/client/models/EvalRequest.java:6]: @markphelps great idea!

File: flipt-client-java/build.gradle (hunk 5 above)
Comment (yquansah) [reply flipt-client-java/build.gradle:87]: Did so! Great idea

**Review by markphelps — APPROVED** [review]
looks great to me!!!

**Pull request merged**